snapshot/snapshot.go

```diff
@@ -403,12 +403,14 @@ func (o *snapshotter) Remove(ctx context.Context, key string) error {
 	}
 
 	if snap.Labels != nil {
-		if imageID, ok := snap.Labels[label.CRIImageRef]; ok {
-			if err := o.fs.DelSnapshot(imageID); err != nil {
-				return errors.Wrap(err, "failed to delete snapshot in cache manager")
+		if _, ok := snap.Labels[label.TargetSnapshotRef]; ok {
+			if imageID, ok := snap.Labels[label.CRIImageRef]; ok {
```

Comment:
Why need to check if a particular label existed for the deleted snapshot?  What's the problem if we delete a snapshot which does not meet the labels combination?

Reply:
Does it work for OCI image layer?

Reply:
As discussed with @power-more , it does not affect the OCI layer.